rest-server/src/model/aog.hpp

```diff
@@ -0,0 +1,47 @@
+//
+// Created by shulce on 1/16/19.
+//
+
+#ifndef REST_SERVER_AOG_HPP
+#define REST_SERVER_AOG_HPP
+
+#endif //REST_SERVER_AOG_HPP
+
+#include <string>
+#include "cpprest/json.h"
+
+namespace cadg_rest {
+
+    struct Aog {
+        int id;
+        std::string name;
+        std::string agency;
+        web::json::value to_json() {
+            auto aog_json = web::json::value::object();
+            aog_json["name"] = web::json::value::string(name);
+            aog_json["agency"] = web::json::value::string(agency);
+            return aog_json;
+        }
+
+        static std::optional<Aog> from_json(web::json::value input) {
+            if (input.has_field("name") && input["name"].is_string() &&
+                input.has_field("agency") && input["agency"].is_string()) {
+                if (input.has_field("id") && input["id"].is_integer()) {
+                    return Aog {input["id"].as_integer(),
```

Comment:
Not needed, integers default value is 0. String default value is empty string

Reply:
in-line construction